### PR TITLE
Autosave mmt when navigating away

### PR DESCRIPTION
**Pull request by annapetry**
This change autosaves a MMT template when a User navigates away from the MMT.edit page.  This eliminates the need for an overlay and for the `AlertUnsavedChanges` mixin.

### CODE REVIEW

File: client/tests/integration/manuscript-manager-template-test.coffee [hunk 1]
```diff
@@ -23,7 +23,10 @@ module 'Integration: Manuscript Manager Templates',
   setup: ->
     app = startApp()
     server = setupMockServer()
-
+    Tahi.__container__.lookup(
+      'controller:admin/journal/manuscript-manager-template/edit'
+    )._actions.saveTemplateOnClick = -> console.log 'No Action'
+
```
Comment (mikem) [inline client/tests/integration/manuscript-manager-template-test.coffee]: Hm, why is this necessary? Is the original function restored after the test finishes?

File: client/tests/integration/manuscript-manager-template-test.coffee [hunk 2]
```diff
@@ -168,4 +171,4 @@ test 'click delete confirmation overlay submit button', ->
     equal find(".card-content").length, 0, "The card is gone"
   andThen ->
     search = { method: "DELETE", url: "/task_templates/1" }
-    ok _.findWhere(server.requests, search), "It sends DELETE request to the server"
+    ok _.findWhere(server.responses, search), "It sends DELETE request to the server"
```
Comment (mikem) [inline client/tests/integration/manuscript-manager-template-test.coffee]: Why was this change necessary?

File: client/tests/integration/manuscript-manager-template-test.coffee (hunk 1 above)
Comment (annapetry) [inline client/tests/integration/manuscript-manager-template-test.coffee]: Good question.  I should restore the function in the teardown. 

I was running into a timing issue after adding the autosave functionality.  Once a test finishes, the session is destroyed in `teardown`.  But by navigating away from the MMT, the teardown triggers an autosave event.  Then the test fails because there are records that are `inFlight`.

File: client/tests/integration/manuscript-manager-template-test.coffee (hunk 2 above)
Comment (annapetry) [inline client/tests/integration/manuscript-manager-template-test.coffee]: When I looked a the `server.requests`, I only saw four `GET` requests.  When I looked at the `server.responses`, I saw all of the stubbed out HTTP requests.  

Is there a better way to do this?

File: client/tests/integration/manuscript-manager-template-test.coffee (hunk 1 above)
Comment (mikem) [inline client/tests/integration/manuscript-manager-template-test.coffee]: To address that, can we end each test with either a save or a cancel click? I'd like to avoid this stubbing if possible.

File: client/tests/integration/manuscript-manager-template-test.coffee (hunk 2 above)
Comment (mikem) [inline client/tests/integration/manuscript-manager-template-test.coffee]: I don't know, I was just curious why this change was made. I'm not that familiar with this API (Sinon, right)?

File: client/tests/integration/manuscript-manager-template-test.coffee (hunk 1 above)
Comment (annapetry) [inline client/tests/integration/manuscript-manager-template-test.coffee]: I couldn't find a better way to handle this.

- I tried sinon again to no avail.
- I tried to preemptively navigate away from the page in order to avoid the request being triggered in `teardown`.  The tests, though, are so quick, so the records were still in flight when `teardown` is eventually called.

File: client/tests/integration/manuscript-manager-template-test.coffee [hunk 3]
```diff
@@ -23,7 +23,12 @@ module 'Integration: Manuscript Manager Templates',
   setup: ->
     app = startApp()
     server = setupMockServer()
-
+    # TODO NEEDS SOME LOVE
+    # Couldn't successfully fo this with sinon
+    Tahi.__container__.lookup(
+      'controller:admin/journal/manuscript-manager-template/edit'
+    )._actions.saveTemplateOnClick = -> console.log 'No Action'
```
Comment (mikem) [inline client/tests/integration/manuscript-manager-template-test.coffee]: We should store and replace the original function in `teardown`.

**Comment by annapetry** [comment]
@mikem Restored the function in teardown.  

My CircleCI builds are failing with many bogus errors, FYI.

**Comment by mikem** [comment]
Looks good to me. :+1:
